seahub/api2/endpoints/admin/web_settings.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2012-2019 Seafile Ltd.
+
+import logging
+from constance import config
+
+from django.conf import settings as dj_settings
+from django.http import Http404
+from django.utils.translation import ugettext as _
+
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAdminUser
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import status
+
+from seahub.api2.authentication import TokenAuthentication
+from seahub.api2.throttling import UserRateThrottle
+from seahub.api2.utils import api_error
+from seahub.utils.auth import get_login_bg_image_path
+logger = logging.getLogger(__name__)
+
+
+class AdminWebSettings(APIView):
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    throttle_classes = (UserRateThrottle,)
+    permission_classes = (IsAdminUser,)
+
+    def get(self, request):
+        if not dj_settings.ENABLE_SETTINGS_VIA_WEB:
+            error_msg = 'Web settings not supported.'
+            return api_error(status.HTTP_404_NOT_FOUND, error_msg)
+
+        config_dict = {}
+        for key in dir(config):
+            value = getattr(config, key)
+            config_dict[key] = value
+
+        return Response(config_dict)
+
+    def put(self, request):
+
+        DIGIT_WEB_SETTINGS = [
+            'DISABLE_SYNC_WITH_ANY_FOLDER', 'ENABLE_SIGNUP',
+            'ACTIVATE_AFTER_REGISTRATION', 'REGISTRATION_SEND_MAIL',
+            'LOGIN_REMEMBER_DAYS', 'REPO_PASSWORD_MIN_LENGTH',
+            'ENABLE_REPO_HISTORY_SETTING', 'USER_STRONG_PASSWORD_REQUIRED',
+            'ENABLE_ENCRYPTED_LIBRARY', 'USER_PASSWORD_MIN_LENGTH',
+            'USER_PASSWORD_STRENGTH_LEVEL', 'SHARE_LINK_PASSWORD_MIN_LENGTH',
+            'ENABLE_USER_CREATE_ORG_REPO', 'FORCE_PASSWORD_CHANGE',
+            'LOGIN_ATTEMPT_LIMIT', 'FREEZE_USER_ON_LOGIN_FAILED',
+            'ENABLE_SHARE_TO_ALL_GROUPS', 'ENABLE_TWO_FACTOR_AUTH',
+            'ENABLE_BRANDING_CSS', 'ENABLE_TERMS_AND_CONDITIONS',
+            'ENABLE_USER_CLEAN_TRASH', 'SHARE_LINK_TOKEN_LENGTH'
+        ]
+
+        STRING_WEB_SETTINGS = ('SERVICE_URL', 'FILE_SERVER_ROOT', 'TEXT_PREVIEW_EXT',
+                               'SITE_NAME', 'SITE_TITLE', 'CUSTOM_CSS')
+
+        res = []
+        for key, value in request.data.items():
+            if key not in dir(config) or value is None:
+                error_msg = _(u'setting invalid.')
+                return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+
+            if value.isdigit():
```

Comment:
if key not in str_s and not in digit_s:
   error


if key in str_setting and value not is str:
    error

if key in digit_setting and value is not digit:
    error
...